rusty_pizza_server/src/order_model/meal.rs

```diff
@@ -0,0 +1,95 @@
+use crate::util::money::Money;
+use std::collections::BTreeSet;
+
+#[derive(Debug, PartialEq, Eq, Hash)]
+pub struct Meal {
+    /// Number of the meal in the menu
+    meal_id: String,
+    /// Size of the pizza or noodle type etc.
+    variety: String,
+    specials: BTreeSet<String>,
+    price: Money,
+}
+
+impl Meal {
+    pub fn add_special(&mut self, special: String) {
+        self.specials.insert(special);
+    }
+
+    pub fn remove_special(&mut self, special: &String) {
+        self.specials.remove(special);
+    }
+
+    pub(super) fn new_for_test(meal_id: String, variety: String, price: Money) -> Meal {
+        Meal {
+            meal_id,
+            variety,
+            price,
+            specials: BTreeSet::new(),
+        }
+    }
```

Comment:
Having this function just to create an instance for tests feels off. "Tests rule" but I think there is a better option. As we already discussed we need to change the way objects are constructed and inserted into the datamodel anyway. In the anticipated version `Meals` would create `Meal`. Therefore there must be a constructor anyway (btw. a [builder](https://doc.rust-lang.org/1.0.0/style/ownership/builders.html) could be a good choice here because specials are optional). So why not make it a `pub` constructor?

Reply:
At the end a builder seems the best solution, we should open a Issue for it.
I will change the test constructors to normal constructors in the scope of `pub(super)`. So all models can use these constructors but from outer scope they are not visible

Reply:
Since Rust has no method overloading, i combined the constructors to one pub constructor for each model

Reply:
Created #10 for this purpose.